app/controllers/declaration/ProcedureCodesPageController.scala

```diff
@@ -209,7 +209,7 @@ class ProcedureCodesPageController @Inject()(
 
       case (procedureCode, seq) =>
         procedureCode match {
-          case ProcedureCodes(None, _) if cachedData.procedureCode.isEmpty =>
+          case ProcedureCodes(None, _) if procedureCode.procedureCode.isEmpty =>
```

Comment:
This is wrong. You're matching procedure code so `ProcedureCodes(None, _)` is exactly the same like `if procedureCode.procedureCode.isEmpty`

Previous approach checked form and cached data

Reply:
If this logic is correct, you don't need `if`. case is enough, but I think it's worth to double check this.

Whole this controller should be refactored, it's huge with really old logic. One of the first pages with multiply item, so approach is not the best

Reply:
Moreover with this approach now, in the first case you will catch everything that don't have `procedureCode`, doesn't matter that you have adiitional procedure code and the same with cached data etc

Reply:
Thanks for spotting this :)